Organize source README (#205)
FuelLabs/sway-libs: README.md

These libraries contain helper functions and other tools valuable to blockchain development.

> > **NOTE:**

Reviewer: Aaah! NOTE! :)

Author: It looks better

#### Access Control and Security

- [Ownership](./libs/ownership/) is used to apply restrictions on functions such that only a single user may call them.
- [Pausable](./libs/pausable/) allows contracts to implement an emergency stop mechanism.
- [Reentrancy](./libs/reentrancy) is used to detect and prevent reentrancy attacks.

#### Cryptography

- [Merkle Proof](./libs/merkle_proof/) is used to verify Binary Merkle Trees computed off-chain.

#### Math

- [Fixed Point Number](./libs/fixed_point/) is an interface to implement fixed-point numbers.
- [Signed Integers](./libs/signed_integers/) is an interface to implement signed integers.

#### Data Structures

- [Queue](./libs/queue/) is a linear data structure that provides First-In-First-Out (FIFO) operations. 

## Using a library

To import a library, a dependency should be added to the project's `Forc.toml` file under `[dependencies]`. The following shows adding the Merkle Proof Library.

```rust
merkle_proof = { git = "https://github.com/FuelLabs/sway-libs", tag = "v0.1.0" }
```

> **NOTE:** 

> > Be sure to set the tag as the most recent release.

Reviewer: ```suggestion
> Be sure to set the tag to the latest release.
```

#### Math

- [Fixed Point Number](./libs/fixed_point/) is an interface to implement fixed-point numbers.
- [Signed Integers](./libs/signed_integers/) is an interface to implement signed integers.

#### Data Structures

- [Queue](./libs/queue/) is a linear data structure that provides First-In-First-Out (FIFO) operations. 

## Using a library

To import a library, a dependency should be added to the project's `Forc.toml` file under `[dependencies]`. The following shows adding the Merkle Proof Library.

```rust
merkle_proof = { git = "https://github.com/FuelLabs/sway-libs", tag = "v0.1.0" }
```

> **NOTE:** 
> Be sure to set the tag to the latest release.

You may then import your desired library in your Sway Smart Contract as so:

```sway
use merkle_proof::<library_function>;
```

For example, to import the `verify_proof()` function use the following statement:

```sway
use merkle_proof::binary_merkle_proof::verify_proof;
```

## Running Tests

There are two sets of tests that should be run: inline tests and sdk-harness tests.

In order to run the inline tests, make sure you are in the `libs/` folder of this repository `sway-libs/libs/<you are here>`.

Run the tests:

```rust

> forc test

Reviewer: Could you amend the `rust` language declaration above to be `bash` and add one to the code block below too?

```

Once these tests have passed, make sure you are in the `tests/` folder of this repository `sway-libs/tests/<you are here>`.

Run the tests:

```
forc test && cargo test
```

> **NOTE:**
> This may take a while depending on your hardware, future improvements to Sway will decrease build times. After this has been run once, indiviual test projects may be built on their own to save time.

Any instructions related to using a specific library should be found within the README.md of that library.

> **NOTE:**

> > All projects currently use `forc v0.47.0`, `fuels-rs v0.46.0` and `fuel-core 0.20.8`.

Reviewer: oh, just wondering about the sdk version here. do the latest releases work with that version? I thought that was out of date and incompatible

Author: I haven't gotten around to updating the SDK. The focus has been on the libraries themselves such that devs can use them for the hackathon. Another reason why I want to do as much inline testing as possible.
